src/main.ts

```diff
@@ -262,6 +262,26 @@ export class Purchases {
     );
   }
 
+  /**
+   * Retrieves a specific offering by its identifier.
+   * @param offeringIdentifier - The identifier of the offering to retrieve.
+   * @param params - The parameters object to customise the offerings fetch. Check {@link GetOfferingsParams}
+   */
+  public async getOffering(
+    offeringIdentifier: string,
+    params?: GetOfferingsParams,
+  ): Promise<Offering | null> {
+    const appUserId = this._appUserId;
+    const offeringsResponse = await this.backend.getOfferings(appUserId);
+
+    const offerings = await this.getAllOfferings(
```

Comment:
I think that this method won't solve the problem we are trying to address.

If you check inside the `getAllOfferings` method [here](https://github.com/RevenueCat/purchases-js/blob/540b45eb6f89136d473d95439ec7f6e7e4bd8142/src/main.ts#L270-L278) you can see that:
1. it extracts the products from the packages of all offerings
2. calls `/products` with all the product_ids from all offerings

We need to call the `/products` endpoint with the product_ids that are only from the offering we are considering.
So we probably need to build an equivalent of what `getAllOfferings` is doing but with a `offering:Offering` parameter so that it will download only the products from that offering.

Reply:
Of course, thanks for catching it 👍